ts/sagas/wallet.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * A saga that manages the Profile.
+ */
+
+import { call, Effect, put, takeLatest } from "redux-saga/effects";
+
+import { WalletAPI } from "../api/wallet/wallet-api";
+import {
+  FETCH_CARDS_REQUEST,
+  FETCH_TRANSACTIONS_REQUEST
+} from "../store/actions/constants";
+import { cardsFetched } from "../store/actions/wallet/cards";
+import { transactionsFetched } from "../store/actions/wallet/transactions";
+import { CreditCard } from "../types/CreditCard";
+import { WalletTransaction } from "../types/wallet";
+
+function* fetchTransactions(): Iterator<Effect> {
+  const transactions: ReadonlyArray<WalletTransaction> = yield call(
+    WalletAPI.getAllTransactions
+  );
+  yield put(transactionsFetched(transactions));
+}
+
+function* fetchCards(): Iterator<Effect> {
+  const cards: ReadonlyArray<CreditCard> = yield call(WalletAPI.getCreditCards);
+  yield put(cardsFetched(cards));
+}
+
+// This function listens for Profile related requests and calls the needed saga.
+export default function* root(): Iterator<Effect> {
+  yield takeLatest(FETCH_TRANSACTIONS_REQUEST, fetchTransactions);
```

Comment:
you need to wrap these with a saga that takes the latest `walletToken` from the state and then builds a client using that token to make the requests, like [here](https://github.com/teamdigitale/italia-app/blob/master/ts/sagas/messages.ts#L212)

Reply:
While I agree that this needs to happen, I would postpone this until [#158068259](https://www.pivotaltracker.com/story/show/158068259) (which is next on my list and already in progress locally). 

The reason for this is that -- since the app still uses dummy data -- it wouldn't be much use to retrieve the token etc. so the previously linked story seems like a more reasonable place where this logic should be implemented.

Also, I'd like to know whether you have already discussed an "updating strategy". There are three options that I can think of:
1. the data is updated when first launching the app/relevant screen: simple, but since cards and transactions can be added from other channel, the app would not always be up to date
2. "pull down to refresh":

Reply:
While I agree that this needs to happen, I would postpone this until [#158068259](https://www.pivotaltracker.com/story/show/158068259) (which is next on my list and already in progress locally). 

The reason for this is that -- since the app still uses dummy data -- it wouldn't be much use to retrieve the token etc. so the previously linked story seems like a more reasonable place where this logic should be implemented.

Also, I'd like to know whether you have already discussed an "updating strategy". There are three options that I can think of:
1. the data is updated when first launching the app/relevant screen: simple, but since cards and transactions can be added from other channel, the app would not always be up to date
2. "pull down to refresh": allows the user to refresh the page upon request, and it's easy on the data consumed, since it only fetches data when requested. The downside of this is that it may not be particularly "enjoyable" from a user's perspective to have to refresh the page "manually"
3. long polling on the PagoPA server: probably what I would consider to be the best option (maybe coupled with (2)), although some considerations on the amount of data consumed by the user and/or the strain on the servers may be in order

If you have already come to a conclusion about this or if you haven't and would like to discuss the matter further let me know!